### ISSUE TITLE
Reproducing results with resnet50

### ISSUE THREAD

**Issue by salvatorepolizzotto** [GanlongZhao/CST_DA_detection#1]
Hi!
Thanks for sharing your code.
I successfull reproduce your results with vgg16 on the Cityscapes to Foggy setting. Now I was wondering if there is some way to reproduce the result with resnet50, if there are some changes in the code that are necessary for running the experiments with this net. 
Thanks  in advance!

**Comment by uitrbn** [comment]
Hi, I haven't test with resnet50, but it seems that our method is not related to the backbone. So I think what you have to do are: 1) fixing some layers of backbone as we did in vgg16. Please refer to https://github.com/VisionLearningGroup/DA_Detection and their paper. 2) modifying some layers in the rest of the model since the shape of features has changed. For example, the input dimension of the local discriminator.

**Comment by salvatorepolizzotto** [comment]
Thanks a lot. I have implemeted the resnet50 like the paper that you have linked above and changed the input of the Local Discriminator and the training is started. Until now all seems to be ok. But, for better clarity, I want to ask you a thing related to the files needed to make the training and evaluation. I have seen that you impose a condition if the net is equal to vgg16.
For example in cityscapes_pretrain.py, line 309, where if net is equal to vgg16 then you perform clip_gradient(..).
With resnet50 is also neccessary to do this operation in those files, like cityscapes_pretrain.py and cityscapes_to_foggycityscapes_da.py?
Thanks a lot for the support

**Comment by uitrbn** [comment]
The clip_gradient is from the original faster RCNN implementation we used, please refer to https://github.com/jwyang/faster-rcnn.pytorch for more details.

**Comment by salvatorepolizzotto** [comment]
Thanks a lot for your availability. 
Sorry for all the requests, but the code to run the experiment KITTY to Cityscapes and Sim10k to Cityscapes is not available?
And another question: in the paper you explain that you first pretrain the model and then you perform 10,000 iterations with L_det calculated on the source domain and L_adv on both domains, and finally train the network with all loss terms for 6,000 iterations. My question is, when i run cityscapes_to_foggycityscapes_da.py the model perfom 6 epoch with 2965 iteration and so the number of total iteration is not the same as you mention in the paper. Can you explain me why?

**Comment by uitrbn** [comment]
Sorry, I am not at the university now, but the code of KITTY and Sim10k is basically the same, except that you have to remove the DontCare bounding boxes in KITTI.
No reason. I just rewrote the code before I released it so it might be slightly different.

**Comment by salvatorepolizzotto** [comment]
Ok thanks, I'll try to code and run also these experiments. If, in these days, you could upload the code then I'll see if all what I'll do will be ok (regarding kitty e sim10k).
Thank again

**Comment by salvatorepolizzotto** [comment]
Hi! 
Sorry again for the noise, but I want to ask you something related to 2 settings of the paper.

1. In the cityscapes to foggy setting did you use all 3 levels of fog for the foggy dataset during the adaptation and the test on the validation set? Because somebody uses only the level with more fog both for the training phase and the testing phase.

2. Another question is regarding kitti to cityscapes setting: during the pretrain and the adaptation did you use only the class car or all classes of kitti equal to the classes of cityscapes or something else?

Thanks in advance

**Comment by uitrbn** [comment]
1. Yes we did use 3 levels of fog following SCDA(https://github.com/xinge008/SCDA/issues/3) because there seem to be some problems in SWDA. Please refer to https://github.com/VisionLearningGroup/DA_Detection/issues/23.

2. We used all classes of kitti equal to the classes of cityscapes.

**Comment by salvatorepolizzotto** [comment]
Thanks again for the support. If it is possible, can you tell me what are specifically the classes used in the setting kitti to cityscape?
Thanks in advance
